tests/containers/helm.pm

```diff
@@ -121,9 +125,31 @@ sub post_fail_hook {
     script_run("kubectl config set-context --current --namespace=default");
     script_run("kubectl delete namespace helm-ns-$job_id");
 
+    helm_cleanup() unless $self->{is_k3s};
     uninstall_k3s() if $self->{is_k3s};
 }
 
+sub helm_cleanup {
+    diag("helm_cleanup started.");
+    my $namespaces_output = script_output("kubectl get namespaces -o=json");
+    my $namespaces_json = decode_json($namespaces_output);
+    my @namespaces = @{$namespaces_json->{items}};
+    foreach my $item (@namespaces) {
+        next if ($item->{metadata}->{name} !~ /^helm-ns-/);
+        if ($item->{metadata}->{creationTimestamp}) {
+            my $age = time() - str2time($item->{metadata}->{creationTimestamp});
+            if ($age > 10800) { # 3 hours
+                diag("The namespace $item->{metadata}->{name} is $age seconds old. It will be removed.");
+                script_run("kubectl delete all --all -n " . $item->{metadata}->{name});
```

Comment:
I am not sure that it is good idea to put something like into the test , imagine that 2/3/5 tests will execute such code in parallel we can catch some not nice bugs in k8s cluster ...

Reply:
> I don't understand. Only 3 hours old namespaces will be deleted. So most of the time none. If none is deleted then the only API call this does is `kubectl get namespaces -o=json` which is pretty minimal.

containers tests are running in really different setup for different versions of SLE , please tell me how many times this code will run during one day ? 

> But there is question if this should be here or in PCW...

IMO this task is 100% match PCW and 0% match test code

Reply:
I agree with Anton, this should be moved to PCW. We had similar discussion for jobs and the conclussions were that the PCW is the best place to do this